src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/SaveTempDataPropertyFilter.cs

```diff
@@ -56,6 +74,21 @@ public void OnActionExecuting(ActionExecutingContext context)
 
             OriginalValues = new Dictionary<PropertyInfo, object>();
 
+            SetPropertyVaules(tempData, Subject);
+        }
+
+        private void SetPropertyVaules(ITempDataDictionary tempData, object subject)
+        {
+            if (PropertyHelpers == null)
+            {
+                return;
+            }
+
+            if (OriginalValues == null)
```

Comment:
Can you move this to `ApplyTempDataChanges`? `OnActionExecuting` already initializes this property.